stock_quant_internal_transfer/wizard/stock_quant_transfer.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class QuantTransferWizard(models.TransientModel):
+    _name = 'quant.transfer.wizard'
+
+    destination_location_id = fields.Many2one(
+        'stock.location',
+        string='Destination Location',
+        required=True,
+    )
+
+    @api.multi
+    def action_stock_quant_transfer(self):
+        context = dict(self._context or {})
+        stock_quant_obj = self.env['stock.quant']
+        stock_move_obj = self.env['stock.move']
+        stock_picking_obj = self.env['stock.picking']
+        stock_picking_type_obj = self.env['stock.picking.type']
+        stock_warehouse_obj = self.env['stock.warehouse']
+
+        active_ids = context.get('active_ids', [])
+
+        quant_ids = stock_quant_obj.browse(active_ids)
+        if not quant_ids:
+            raise UserError(_('Please select quant(s) for the internal '
+                              'transfer.'))
+
+        source_location = quant_ids[0].location_id
+
+        #  if selected quants are from different location then raise to
+        #  avoid confusion of taking source location for picking
+        if any(q.location_id != source_location for q in quant_ids):
+            raise UserError(_('Please select quants that are in the same '
+                              'Location.'))
+
+        stock_move_list = stock_move_obj.search([
+            ('state', 'not in', ['done', 'cancel']),
+            ('picking_id', '!=', False),
+            ('product_id', 'in', [q.product_id.id for q in quant_ids])
+        ])
+        #  if any quant is already transferred from selected quant then raise
+        if stock_move_list:
+            raise UserError(_('Some of the selected quants are being '
+                              'transferred internally.'))
```

Comment:
"Some of the selected quants are being used in another stock move."  i.e. you cannot say a quant is being transferred *internally* when it is used in an outgoing delivery.

We should also show the names of the products.  It will be hard for users to identify the problematic quants, otherwise.